src/main/java/jenkins/plugins/purgejobhistory/PurgeJobHistoryAction.java

```diff
@@ -93,14 +95,17 @@ public String getUrlName() {
     /**
      * Does the actual purging when triggered via the UI.
      *
-     * @param request        the request.
      * @param resetNextBuild the reset flag.
+     * @param forceDelete force deleting even runs that have been flagged for retention.
      * @return the response.
      * @throws IOException if something goes wrong.
      */
     @RequirePOST
-    public HttpResponse doDoPurge(@QueryParameter boolean resetNextBuild) throws IOException {
-        PurgeJobHistory.purge(job, resetNextBuild);
+    @Restricted(NoExternalUse.class) // only used via stapler
+    public HttpResponse doDoPurge(@QueryParameter("resetNextBuild") boolean resetNextBuild, 
+                                  @QueryParameter("forceDelete") boolean forceDelete)
```

Comment:
You would think.... but I spent 30 minutes with a debugger trying to figure out why the force was not coming through!